packages/flutter/lib/src/material/typography.dart

```diff
@@ -6,623 +6,449 @@ import 'package:flutter/foundation.dart';
-        fontFamily: fontFamily,
-        fontSizeFactor: fontSizeFactor,
-        fontSizeDelta: fontSizeDelta,
-      ),
-      body2: body2.apply(
-        color: bodyColor,
-        decoration: decoration,
-        decorationColor: decorationColor,
-        decorationStyle: decorationStyle,
-        fontFamily: fontFamily,
-        fontSizeFactor: fontSizeFactor,
-        fontSizeDelta: fontSizeDelta,
-      ),
-      body1: body1.apply(
-        color: bodyColor,
-        decoration: decoration,
-        decorationColor: decorationColor,
-        decorationStyle: decorationStyle,
-        fontFamily: fontFamily,
-        fontSizeFactor: fontSizeFactor,
-        fontSizeDelta: fontSizeDelta,
-      ),
-      caption: caption.apply(
-        color: displayColor,
-        decoration: decoration,
-        decorationColor: decorationColor,
-        decorationStyle: decorationStyle,
-        fontFamily: fontFamily,
-        fontSizeFactor: fontSizeFactor,
-        fontSizeDelta: fontSizeDelta,
-      ),
-      button: button.apply(
-        color: bodyColor,
-        decoration: decoration,
-        decorationColor: decorationColor,
-        decorationStyle: decorationStyle,
-        fontFamily: fontFamily,
-        fontSizeFactor: fontSizeFactor,
-        fontSizeDelta: fontSizeDelta,
-      ),
+    return Typography(
+      black: black ?? this.black,
+      white: white ?? this.white,
+      englishLike: englishLike ?? this.englishLike,
+      dense: dense ?? this.dense,
+      tall: tall ?? this.tall,
     );
   }
 
-  /// Linearly interpolate between two text themes.
-  ///
-  /// The arguments must not be null.
+  /// Linearly interpolate between two [Typography] objects.
   ///
   /// {@macro flutter.material.themeData.lerp}
-  static TextTheme lerp(TextTheme a, TextTheme b, double t) {
-    assert(a != null);
-    assert(b != null);
-    assert(t != null);
-    return TextTheme(
-      display4: TextStyle.lerp(a.display4, b.display4, t),
-      display3: TextStyle.lerp(a.display3, b.display3, t),
-      display2: TextStyle.lerp(a.display2, b.display2, t),
-      display1: TextStyle.lerp(a.display1, b.display1, t),
-      headline: TextStyle.lerp(a.headline, b.headline, t),
-      title: TextStyle.lerp(a.title, b.title, t),
-      subhead: TextStyle.lerp(a.subhead, b.subhead, t),
-      body2: TextStyle.lerp(a.body2, b.body2, t),
-      body1: TextStyle.lerp(a.body1, b.body1, t),
-      caption: TextStyle.lerp(a.caption, b.caption, t),
-      button: TextStyle.lerp(a.button, b.button, t),
+  static Typography lerp(Typography a, Typography b, double t) {
+    return Typography(
+      black: TextTheme.lerp(a.black, b.black, t),
+      white: TextTheme.lerp(a.white, b.white, t),
+      englishLike: TextTheme.lerp(a.englishLike, b.englishLike, t),
+      dense: TextTheme.lerp(a.dense, b.dense, t),
+      tall: TextTheme.lerp(a.tall, b.tall, t),
     );
   }
 
   @override
-  bool operator ==(dynamic other) {
+  bool operator ==(Object other) {
```

Comment:
Interesting, we mostly use `dynamic` but a few use `Object`. We really should change them all to `Object`. Not in this PR, obviously...